shared/src/domain-logic/opportunityFields.ts

```diff
@@ -42,3 +57,15 @@ export const isFieldAnswered = (field: keyof ProfileData, data: ProfileData): bo
     }
   }
 };
+
+const areForeignNexusProfileFieldsAnswered = (field: keyof ProfileData, data: ProfileData): boolean => {
+  if (field === "dateOfFormation") return !!data.dateOfFormation;
+
+  if (data.nexusLocationInNewJersey) {
+    if (field === "municipality") return data.municipality !== undefined;
+  } else {
+    if (field === "homeBasedBusiness") return data.homeBasedBusiness !== undefined;
+  }
```

Comment:
Poppy and Dakota have the 5 fields we are checking, for Dakota Nexus with land in NJ, it has dateOfFormation and municipality. For Dakota Nexus with no land in NJ, it has dateOfFormation and homeBaseBusiness.  The ownershipIds and existing employees dont show for either.

Reply:
Got it, I didn't realize that, and I think that learning this only underscores my point about the distinction between `isQuestionAnswered` and `getFieldsForProfile`.

This code is "working" in that the fields are showing up correctly, but only because the logic is saying that for a field like "existingEmployees" for nexus - it is in the array returned from `getFieldsForProfile`. And then when we map over that array to find out if each question is answered, it hits the default `return true` on line 56. Which is saying, logically: "this field exists in the Nexus profile but it has already been answered" which is two statements that are both untrue. It's creating the correct outcome in that it does not show in the alert, sure, but the logic that's causing that outcome is not correct.

Reply:
got it